demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1685 @@
+    otaAgentTask( pParam );
+    LogInfo( ( "OTA Agent stopped." ) );
+    return NULL;
+}
+/*-----------------------------------------------------------*/
+static int startOTADemo( void )
+{
+    /* Status indicating a successful demo or not. */
+    int returnStatus = EXIT_SUCCESS;
+
+    /* coreMQTT library return status. */
+    MQTTStatus_t mqttStatus = MQTTBadParameter;
+
+    /* OTA library return status. */
+    OtaErr_t otaRet = OtaErrNone;
+
+    /* OTA Agent state returned from calling OTA_GetAgentState.*/
+    OtaState_t state = OtaAgentStateStopped;
+
+    /* OTA event message used for sending event to OTA Agent.*/
+    OtaEventMsg_t eventMsg = { 0 };
+
+    /* OTA library packet statistics per job.*/
+    OtaAgentStatistics_t otaStatistics = { 0 };
+
+    /* OTA Agent thread handle.*/
+    pthread_t threadHandle;
+
+    /* OTA interface context required for library interface functions.*/
+    OtaInterfaces_t otaInterfaces;
+
+    /* Maximum time to wait for the OTA agent to get suspended. */
+    int16_t suspendTimeout;
+
+    /* Set OTA Library interfaces.*/
+    setOtaInterfaces( &otaInterfaces );
+
+    LogInfo( ( "OTA over MQTT demo, Application version %u.%u.%u",
+               appFirmwareVersion.u.x.major,
+               appFirmwareVersion.u.x.minor,
+               appFirmwareVersion.u.x.build ) );
+
+    /****************************** Init OTA Library. ******************************/
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        if( ( otaRet = OTA_Init( &otaBuffer,
+                                 &otaInterfaces,
+                                 ( const uint8_t * ) ( CLIENT_IDENTIFIER ),
+                                 otaAppCallback ) ) != OtaErrNone )
+        {
+            LogError( ( "Failed to initialize OTA Agent, exiting = %u.",
+                        otaRet ) );
+
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    /****************************** Create OTA Task. ******************************/
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        if( pthread_create( &threadHandle, NULL, otaThread, NULL ) != 0 )
+        {
+            LogError( ( "Failed to create OTA thread: "
+                        ",errno=%s",
+                        strerror( errno ) ) );
+
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    /****************************** OTA Demo loop. ******************************/
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Wait till OTA library is stopped, output statistics for currently running
+         * OTA job */
+        while( ( ( state = OTA_GetState() ) != OtaAgentStateStopped ) )
+        {
+            if( mqttSessionEstablished != true )
+            {
+                /* Connect to MQTT broker and create MQTT connection. */
+                returnStatus = establishConnection();
+
+                if( returnStatus == EXIT_SUCCESS )
+                {
+                    mqttSessionEstablished = true;
```

Comment:
`establishConnection` already sets this flag to true

Reply:
Thanks, addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/11afb1554eb4fcb7cbaf537332c53d27b1085317